### ISSUE TITLE
Create function - the planned state does not match the desired state after applying

### ISSUE THREAD

**Issue by NathanBaulch** [ariga/atlas#3200]
Something seems to have changed in v0.28.0 - I can no longer apply my schema which contains functions with table row args.

```
> atlas.exe schema apply --env local
Abort: the planned state does not match the desired state after applying the file:
```

```diff
--- planned state
+++ desired state
@@ -18,7 +18,7 @@
 );
 CREATE INDEX "ix__contact__name-id" ON "contact" ("name", "id");
 CREATE INDEX "ix__contact__resource_groups-id" ON "contact" ("resource_groups", "id");
-CREATE FUNCTION "get_contact_tsv" ("data" "contact") RETURNS tsvector LANGUAGE plpgsql AS $$
+CREATE FUNCTION "get_contact_tsv" ("data" myapp.contact) RETURNS tsvector LANGUAGE plpgsql AS $$
 BEGIN
     RETURN setweight(to_tsvector(coalesce(data.name, '')), 'A');
 END;
```

Here's the function definition. I've also tried specifying the arg with `type = table.contact` but no change.

```hcl
function "get_contact_tsv" {
  schema = schema.myapp
  lang   = PLpgSQL
  arg "data" {
    type = sql("contact")
  }
  return = tsvector
  as     = <<-SQL
  BEGIN
      RETURN setweight(to_tsvector(coalesce(data.name, '')), 'A');
  END;
  SQL
  depends_on = [table.contact]
}
```

I've also tried with v0.28.1 without luck. Where am I going wrong?

**Comment by a8m** [comment]
Hey @NathanBaulch 👋 

Thanks for reporting this. I tried to reproduce it locally without success. Here is the schema I used:

```hcl
schema "myapp" {}

table "contact" {
  schema = schema.myapp
  column "name" {
    type = text
  }
}

function "get_contact_tsv" {
  schema = schema.myapp
  lang   = PLpgSQL
  arg "data" {
    type = table.contact
  }
  return = tsvector
  as     = <<-SQL
  BEGIN
      RETURN setweight(to_tsvector(coalesce(data.name, '')), 'A');
  END;
  SQL
}
```

Just note, I used `type = table.contact` instead, since `myapp.contact` may differs from `contact` as the default `search_path` is `public`.

Could you provide a bit more information to help reproduce this? Are you using a `dev-url` in this env? Can you share the `dev-url`/`url` value you are using (omit the user/password/hostname before sharing).

**Comment by a8m** [comment]
OK, I was able to reproduce this when I scoped my connection to a specific schema.

The issue related to the new lint stage that was added to the declarative workflow (`schema plan` and `schema apply`) and it was fixed with #3201.

Thanks for reporting this.
